src/shared/components/palette/fixed-list/write-fixed-list-field.component.ts

```diff
@@ -8,17 +8,13 @@ import { AbstractFieldWriteComponent } from '../base-field/abstract-field-write.
 })
 export class WriteFixedListFieldComponent extends AbstractFieldWriteComponent implements OnInit {
 
-  fixedListControl: FormControl;
+  fixedListFormControl: FormControl;
 
   ngOnInit() {
-    let notEmpty = this.caseField.value !== null && this.caseField.value !== undefined;
-    this.fixedListControl = this.registerControl(new FormControl(notEmpty ? this.caseField.value : ''));
-    if (this.hasSanitisedDynamicListData()) {
-      this.fixedListControl.setValue(this.caseField.value);
+    let isNull = this.caseField.value === undefined || this.caseField.value === '';
+    if (isNull) {
+      this.caseField.value = null;
     }
-  }
-
-  private hasSanitisedDynamicListData() {
-    return this.fixedListControl.value && this.fixedListControl.value.value;
```

Comment:
What is the reason for change of logic here? Why are we not checking for `value.value`?

Reply:
@kiran-yenigala-hmcts Please could you address Andrew's query? ⬆️

Reply:
We have changed implementation of this component to set to default value of --Select a value-- if caseField.value is not set of caseField.value == null. And to return null in the result when --Select a value-- is selected.
To address above issues, in the html file we have changed to [(ngModel)] implementation which is now bounded to caseField.value.
And in the ts file, we are just checking for undefined or '' (empty) value on the caseField.value, if that is true then sets caseField.value to null so that on the UI it selectes --Select a value--.

Reply:
Was it tested with dynamic list? The value can be:
```"DynamicList": {
  "value": {
    "code": "List1",
    "label": " List 1"
  },
  "list_items": [{
    "code": "List1",
    "label": " List 1"
  }, {
    "code": "List2",
    "label": " List 2"
  }, {
    "code": "List3",
    "label": " List 3"
  }, {
    "code": "List4",
    "label": " List 4"
  }, {
    "code": "List5",
    "label": " List 5"
  }, {
    "code": "List6",
    "label": " List 6"
  }, {
    "code": "List7",
    "label": " List 7"
  }
  ]
}```
hence the `value.value`.

Reply:
@kiran-yenigala-hmcts let me know

Reply:
We have tested on UI with dynamic lists... Ashley has done the initial testing on this..

Reply:
Ok cheers.